### PR TITLE
[Bugfix] Don't use QFileDialog.Options() in _open_folder_dialog

### LINKED ISSUE #6060: [Qt6, PyQt6] AttributeError: type object 'QFileDialog' has no attribute 'Options'
## 🐛 Bug

The dialog to load a folder fails to open.

## To Reproduce

1. Open napari
2. Hit File > Open Folder...

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File ~/miniconda/envs/napari-dlc/lib/python3.10/site-packages/napari/_qt/qt_viewer.py:910, in QtViewer._open_folder_dialog(self=<napari._qt.qt_viewer.QtViewer object>, choose_plugin=False)
    900 hist = get_open_history()
    901 dlg.setHistory(hist)
    903 folder = dlg.getExistingDirectory(
    904     self,
    905     trans._('Select folder...'),
    906     hist[0],  # home dir by default
    907     (
    908         QFileDialog.DontUseNativeDialog
    909         if in_ipython()
--> 910         else QFileDialog.Options()
        dlg = <PyQt6.QtWidgets.QFileDialog object at 0x288e3dcf0>
        self = <napari._qt.qt_viewer.QtViewer object at 0x284db70a0>
        trans = <napari.utils.translations.TranslationBundle object at 0x1309850c0>
        hist = ['/Users/jessy']
        hist[0] = '/Users/jessy'
        QFileDialog = <class 'PyQt6.QtWidgets.QFileDialog'>
        QFileDialog.DontUseNativeDialog = <Option.DontUseNativeDialog: 8>
    911     ),
    912 )
    914 if folder not in {'', None}:
    915     self._qt_open([folder], stack=False, choose_plugin=choose_plugin)

AttributeError: type object 'QFileDialog' has no attribute 'Options'
```

## Expected behavior

The dialog window opens.

## Environment

napari==0.4.18
PyQt6==6.5.1

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# Fixes/Closes

Closes #6060 

# Description

PyQt6 doesn't support QFileDialog.Options(), so this is an analogous change to ` _open_folder_dialog()` as was implemented for `_open_files...` here https://github.com/napari/napari/pull/5492

# References
https://github.com/napari/napari/pull/5492

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] example: all tests pass with my change
The issue is fixed locally, using a test .zarr

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

### CODE REVIEW

File: napari/_qt/qt_viewer.py [hunk 1]
```diff
@@ -835,16 +835,20 @@ def _open_folder_dialog(self, choose_plugin=False):
...
-        )
+        open_kwargs = {
+            "parent": self,
+            "caption": trans._('Select folder...'),
+        }
+        if "pyside" in QFileDialog.__module__.lower():
+            # PySide6
+            open_kwargs["dir"] = hist[0]
+        else:
+            open_kwargs["directory"] = hist[0]
```
Comment (Czaki) [inline napari/_qt/qt_viewer.py:858]: Why this change is required?

Comment (psobolewskiPhD) [reply napari/_qt/qt_viewer.py:858]: I was mimicking the dialog that works above, but I can test keeping/using the positional args instead.

**Comment by psobolewskiPhD** [comment]
Making WIP because there's also the Save dialogs that need to be fixed.

**Comment by psobolewskiPhD** [comment]
This was fixed in qtpy:
https://github.com/spyder-ide/qtpy/releases/tag/v2.4.0
